components/Modal/CreateBlueprint.js

```diff
@@ -105,26 +108,45 @@ class CreateBlueprintModal extends React.Component {
   }
 
   dismissErrors() {
-    this.setState({ errorNameEmpty: false, errorNameDuplicate: false, errorInline: false });
+    this.setState({
+      errorNameEmpty: false,
+      errorNameDuplicate: false,
+      errorNameSpace: false,
+      errorInline: false,
+      errorNameInvalid: false,
+      errorNameInvalidChar: ""
+    });
   }
 
-  handleErrors(blueprintName) {
-    this.handleErrorDuplicate(blueprintName);
-    this.handleErrorName(blueprintName);
-  }
+  handleErrorNameInvalid(blueprintName) {
+    if (blueprintName.length === 0 && this.state.checkErrors) {
+      this.setState({ errorNameEmpty: true });
+    } else {
+      const invalidCharsRegex = /[^a-zA-Z0-9_,.:+*-]/;
+      const nameInvalidChars = blueprintName.search(invalidCharsRegex);
+      const nameContainsSpace = /\s/.test(blueprintName);
 
-  handleErrorDuplicate(blueprintName) {
-    const nameNoSpaces = blueprintName.replace(/\s+/g, "-");
-    if (this.props.blueprintNames.includes(nameNoSpaces)) {
-      this.setState({ errorNameDuplicate: true });
+      if (nameInvalidChars !== -1 && !nameContainsSpace) {
+        this.setState({ errorNameInvalid: true });
+        this.setState({ errorNameInvalidChar: blueprintName[nameInvalidChars] });
+      } else if (nameContainsSpace) {
+        this.setState({ errorNameSpace: true });
+      } else if (this.props.blueprintNames.includes(blueprintName)) {
+        this.setState({ errorNameDuplicate: true });
+      }
     }
   }
 
-  handleErrorName(blueprintName) {
-    if (blueprintName === "" && this.state.checkErrors) {
-      setTimeout(() => {
-        this.setState({ errorNameEmpty: true });
-      }, 200);
+  nameContainsError() {
+    if (
+      this.state.errorNameEmpty ||
```

Comment:
It's not nice to return true/false confitionaly
Just return the result of the condition inside the if